Update oracle adapter docs (#5798)
dbt-labs/docs.getdbt.com: website/docs/docs/core/connect-data-platform/oracle-setup.md

export DBT_ORACLE_USER=<username>
export DBT_ORACLE_PASSWORD=***
export DBT_ORACLE_SCHEMA=<username>
export DBT_ORACLE_DATABASE=example_db2022adb
```

> database name can be retrieved using the following query

Reviewer: ```suggestion
Use the following query to retrieve the database name:
```

</Tabs>

:::info Note
Starting with `dbt-oracle==1.0.2`, it is **optional** to set the `database` name in `profile.yml`

Starting with `dbt-oracle==1.8.0` database key in `profile.yml` is **still optional for all but one** of the dbt-oracle workflows.
if `database` is missing in `profile.yml` the generated catalog used for project documentation will be empty.

From `dbt-oracle==1.8`, we detect that `database` key is missing from `profile.yml` and issue a warning to add it for catalog generation. The warning message also shows the database name that dbt-oracle expects. That way users don't have to worry about "what" the database name is and "how" to get it.
:::

> ### Quoting Configuration

Reviewer: ```suggestion
### Quoting configuration
```

```yaml
quoting:
  database: false
  identifier: false
  schema: false
```
</File>

This is recommended and works for most cases.

### Approximate relation match error

Often users have complained about an approximate relation match as shown below:

```
Compilation Error in model <model>
19:09:40    When searching for a relation, dbt found an approximate match. Instead of guessing
19:09:40    which relation to use, dbt will move on. Please delete <model>, or rename it to be less ambiguous.
  Searched for: <model>
```

This is reported in multiple channels:

- [StackOverFlow Approximate relation Match Error](https://stackoverflow.com/questions/75892325/approximate-relation-match-with-dbt-on-oracle)

- [Github Issue #51](https://github.com/oracle/dbt-oracle/issues/51)

- [Github Issue #143](https://github.com/oracle/dbt-oracle/issues/143)

- [Github Issue #144](https://github.com/oracle/dbt-oracle/issues/144)

In all cases, the solution was to enable quoting only for the database.

To solve this issue of `approximate match` use the following quoting configuration

<File name='dbt_project.yaml'>

```yaml
quoting:
  database: true
```
</File>

<VersionBlock firstVersion="1.5.1"> 

> ## Python models using Oracle Autonomous Database (ADB-S)

Reviewer: hey @aosingh , thanks so much for opening this up and looks great! this section here is wrapped in a version block. since we only have v1.5 and higher on the docs site now, we can remove the versionblock now since it applies to all. wanted to get your thoughts on that?

Author: Sure, we can remove the version block. It makes sense

Author: Let me remove the version block and update the PR

Reviewer: great stuff! removed the versionblocks for you @aosingh !
